lib/lambda-worker/lambda-worker.ts

```diff
@@ -219,4 +206,89 @@ export class LambdaWorker extends cdk.Construct {
     queueMessagesVisable.addAlarmAction(alarmAction);
     queueMessagesVisable.addOkAction(alarmAction);
   }
+
+  isContainerLambda(props: LambdaWorkerProps): boolean {
+    if (
+      props.lambdaProps.dockerImageTag &&
+      props.lambdaProps.ecrRepositoryArn &&
+      props.lambdaProps.ecrRepositoryName &&
+      !props.lambdaProps.entry &&
+      !props.lambdaProps.handler
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  isFunctionLambda(props: LambdaWorkerProps): boolean {
+    if (
+      !props.lambdaProps.dockerImageTag &&
+      !props.lambdaProps.ecrRepositoryArn &&
+      !props.lambdaProps.ecrRepositoryName &&
+      props.lambdaProps.entry &&
+      props.lambdaProps.handler
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  createLambdaFunction(props: LambdaWorkerProps): lambda.Function {
+    if (this.isContainerLambda(props)) {
+      const imageTag: string = props.lambdaProps.dockerImageTag
+        ? props.lambdaProps.dockerImageTag
+        : "";
+      const ecrRepositoryArn: string = props.lambdaProps.ecrRepositoryArn
+        ? props.lambdaProps.ecrRepositoryArn
+        : "";
+      const ecrRepositoryName: string = props.lambdaProps.ecrRepositoryName
+        ? props.lambdaProps.ecrRepositoryName
+        : "";
+      const ecrRepository: IRepository =
+        ecr.Repository.fromRepositoryAttributes(this, `${props.name}-ecr`, {
+          repositoryArn: ecrRepositoryArn,
+          repositoryName: ecrRepositoryName,
+        });
+
+      return new lambda.DockerImageFunction(this, props.name, {
+        code: lambda.DockerImageCode.fromEcr(ecrRepository, {
+          tag: imageTag,
+        }),
+        functionName: props.name,
+        description: props.lambdaProps.description,
+        environment: props.lambdaProps.environment,
+        memorySize: props.lambdaProps.memorySize,
+        reservedConcurrentExecutions:
+          props.lambdaProps.reservedConcurrentExecutions,
+        retryAttempts: props.lambdaProps.retryAttempts,
+        securityGroup: props.lambdaProps.securityGroup,
+        timeout: props.lambdaProps.timeout,
+        vpc: props.lambdaProps.vpc,
+        vpcSubnets: props.lambdaProps.vpcSubnets,
+      });
```

Comment:
Could these be moved to inside factory functions, `return this.createContainerLambda(props);` and `return this.createFunctionLambda(props)`, in order to make `createLambdaFunction` shorter?